apsuite/optics_analysis/tune_correction.py

```diff
@@ -112,12 +106,26 @@ def _get_parameter(self, model=None):
         """."""
         if model is None:
             model = self.model
-        linopt, *_ = self._optics_func(
-            accelerator=model, indices='open')
+        linopt, *_ = self._optics_func(model, indices='open')
         if self.type_optics == self.OPTICS.EdwardsTeng:
             nu1 = linopt.mu1[-1]
             nu2 = linopt.mu2[-1]
         else:
             nu1 = linopt.mux[-1]
             nu2 = linopt.muy[-1]
         return np.array([nu1, nu2])/2/np.pi
+
+    def _optics_func(self, model, indices):
+        if self.type_optics == self.OPTICS.EdwardsTeng:
+            _func = pyaccel.optics.calc_edwards_teng
+        elif self._type_optics == self.OPTICS.Twiss:
+            _func = pyaccel.optics.calc_twiss
+
+        cav = model.cavity_on
+        rad = model.radiation_on
+        model.cavity_on = False
+        model.radiation_on = False
+        ret = _func(accelerator=model, indices=indices)
+        model.cavity_on = cav
```

Comment:
I was thinking that maybe it would be better not to include these cavity & radiation modifications to the tune correction class directly, but rather to the LOCO analysis code, immediately before where tune correction is used.

Performing those changes in tune correction is basically that behavior of manipulating the model without the user's knowledge that we are trying to abandon throughout our codes.

What do you guys think?

Reply:
I agree!

Reply:
Done!